core/src/main/java/com/linecorp/armeria/client/AbstractHttpRequestHandler.java

```diff
@@ -197,7 +202,9 @@ final boolean tryInitialize() {
     final void writeHeaders(RequestHeaders headers) {
         final SessionProtocol protocol = session.protocol();
         assert protocol != null;
-        if (headersOnly) {
+        if (shouldExpect100ContinueHeader(headers)) {
```

Comment:
Should we raise an exception or silently ignore the header when the request body is empty?
`A client MUST NOT generate a 100-continue expectation in a request that does not include content.`
https://datatracker.ietf.org/doc/html/rfc9110#section-10.1.1